static/app/views/issueDetails/groupSimilarIssues/similarStackTrace/toolbar.tsx

```diff
@@ -32,27 +60,66 @@ class SimilarToolbar extends Component<Props, State> {
     }
 
     if (mergeList.length !== this.state.mergeCount) {
-      this.setState({mergeCount: mergeList.length});
+      this.setState({mergeCount: mergeList.length, mergeList});
     }
   };
 
   listener = GroupingStore.listen(this.onGroupChange, undefined);
 
   render() {
-    const {onMerge} = this.props;
-    const {mergeCount} = this.state;
+    const {onMerge, parentGroupId, organization} = this.props;
+    const {mergeCount, mergeList} = this.state;
+    const hasSimilarityEmbeddingsFeature = organization?.features?.includes(
+      'issues-similarity-embeddings'
+    );
 
     return (
       <PanelHeader hasButtons>
-        <Confirm
-          disabled={mergeCount === 0}
-          message={t('Are you sure you want to merge these issues?')}
-          onConfirm={onMerge}
-        >
-          <Button size="xs" title={t('Merging %s issues', mergeCount)}>
-            {t('Merge %s', `(${mergeCount || 0})`)}
-          </Button>
-        </Confirm>
+        <ButtonPanel>
+          <Confirm
+            disabled={mergeCount === 0}
+            message={t('Are you sure you want to merge these issues?')}
+            onConfirm={onMerge}
+          >
+            <Button size="xs" title={t('Merging %s issues', mergeCount)}>
+              {t('Merge %s', `(${mergeCount || 0})`)}
+            </Button>
+          </Confirm>
+          {hasSimilarityEmbeddingsFeature && (
+            <React.Fragment>
```

Comment:
avoid importing all of react
```suggestion
            <Fragment>
```